src/components/MultiGestureCanvas/usePanGesture.js

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable no-param-reassign */
+import {Gesture} from 'react-native-gesture-handler';
+import {useDerivedValue, useSharedValue, withDecay, withSpring} from 'react-native-reanimated';
+import * as MultiGestureCanvasUtils from './utils';
+
+const PAN_DECAY_DECELARATION = 0.9915;
+
+const usePanGesture = ({canvasSize, contentSize, zoomScale, totalScale, offsetX, offsetY, panTranslateX, panTranslateY, isSwipingInPager, stopAnimation}) => {
+    // The content size after fitting it to the canvas and zooming
+    const zoomedContentWidth = useDerivedValue(() => contentSize.width * totalScale.value, [contentSize.width]);
+    const zoomedContentHeight = useDerivedValue(() => contentSize.height * totalScale.value, [contentSize.height]);
+
+    // Pan velocity to calculate the decay
+    const panVelocityX = useSharedValue(0);
+    const panVelocityY = useSharedValue(0);
+
+    // Calculates bounds of the scaled content
+    // Can we pan left/right/up/down
+    // Can be used to limit gesture or implementing tension effect
+    const getBounds = MultiGestureCanvasUtils.useWorkletCallback(() => {
+        let horizontalBoundary = 0;
+        let verticalBoundary = 0;
+
+        if (canvasSize.width < zoomedContentWidth.value) {
+            horizontalBoundary = Math.abs(canvasSize.width - zoomedContentWidth.value) / 2;
+        }
+
+        if (canvasSize.height < zoomedContentHeight.value) {
+            verticalBoundary = Math.abs(zoomedContentHeight.value - canvasSize.height) / 2;
+        }
+
+        const horizontalBoundaries = {min: -horizontalBoundary, max: horizontalBoundary};
+        const verticalBoundaries = {min: -verticalBoundary, max: verticalBoundary};
+
+        const clampedOffset = {
+            x: MultiGestureCanvasUtils.clamp(offsetX.value, horizontalBoundaries.min, horizontalBoundaries.max),
+            y: MultiGestureCanvasUtils.clamp(offsetY.value, verticalBoundaries.min, verticalBoundaries.max),
+        };
+
+        // If the horizontal/vertical offset is the same after clamping to the min/max boundaries, the content is within the boundaries
+        const isInHoriztontalBoundary = clampedOffset.x === offsetX.value;
+        const isInVerticalBoundary = clampedOffset.y === offsetY.value;
+
+        return {
+            horizontalBoundaries,
+            verticalBoundaries,
+            clampedOffset,
+            isInHoriztontalBoundary,
+            isInVerticalBoundary,
+        };
+    }, [canvasSize.width, canvasSize.height]);
+
+    // We want to smoothly gesture by phasing out the pan animation
+    // In case the content is outside of the boundaries of the canvas,
+    // we need to return to the view to the boundaries
```

Comment:
> we need to return to the view to the boundaries

Does this have an extra "to" in it somewhere? I don't think I quite understand what it means.